comments/comments.go

```diff
@@ -1,33 +1,32 @@
-package main
+package comments
 
 import (
-	"errors"
+	"bytes"
 	"fmt"
 	"go/ast"
+	"go/format"
 	"go/parser"
 	"go/printer"
 	"go/token"
-	"log"
-	"os"
+	"io/ioutil"
 	"sort"
 	"strings"
 )
 
-func main() {
-	fset := token.NewFileSet()
-	if len(os.Args) == 0 {
-		log.Fatal(errors.New("args is too short"))
-	}
-	fname := os.Args[1]
-	_, err := os.Stat(fname)
-	if err != nil {
-		log.Fatal(errors.New("file not exists"))
+//Process is adding comment templates to should-commented variable names, structure definitions, and function names
+func Process(filename string, src []byte) ([]byte, error) {
+	if src == nil {
+		b, err := ioutil.ReadFile(filename)
+		if err != nil {
+			return nil, err
+		}
+		src = b
 	}
-	f, err := parser.ParseFile(fset, fname, nil, parser.ParseComments)
+	fset := token.NewFileSet()
+	f, err := parser.ParseFile(fset, filename, src, parser.ParseComments)
 	if err != nil {
-		log.Fatal(err)
+		return nil, err
 	}
-
 	comments := []*ast.CommentGroup{}
```

Comment:
The maximum length of `comments` is `len(f.Comments)`,
so you can write `comments := make([]*ast.CommentGroup, 0, len(f.Comments))`